include/openenclave/corelibc/bits/jmp_buf.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* These are the registers that are preserved across function calls
+ * according to the 'System V AMD64 ABI' calling conventions:
+ * RBX, RSP, RBP, R12, R13, R14, R15. In addition, oe_setjmp() saves
+ * the RIP register (instruction pointer) to know where to jump back to).
+ */
+typedef struct ___OE_JMP_BUF
+{
+    uint64_t rsp;
+    uint64_t rbp;
+    uint64_t rip;
+    uint64_t rbx;
+    uint64_t r12;
+    uint64_t r13;
+    uint64_t r14;
+    uint64_t r15;
+    /* Added these to align with size of MUSL jmp_buf */
+    uint64_t __fl;
+    uint64_t __ss[128 / sizeof(long)];
+} __OE_JMP_BUF;
```

Comment:
Additionally, the control bits of the MXCSR register and the x87 control word are preserved across calls i.e. callee-saved. Let's add this info to the comments section.
Also the direction flag DF in the RFlags register must be clear (set to "forward" direction) on function entry and return i.e. preserved.

Reply:
We didn't write this code. We just moved it to a different header. This should be tracked (as a separate) as a separate issue.

Reply:
ok, sounds good.